tests/unit/facet.spec.js

```diff
@@ -2,11 +2,61 @@ import { shallowMount } from "@vue/test-utils";
 import Facet from "@/components/Facet.vue";
 
 describe("Facet.vue", () => {
-  it("renders props.facet when passed", () => {
-    const facet = "new message";
+  it("displays header and available facets in sidebar", () => {
+    const mockRoute = {
+      query: { q: "cheese", page: "1" },
+    };
+    const mockRouter = {
+      push: jest.fn(),
+    };
+
+    const wrapper = shallowMount(Facet, {
+      global: {
+        mocks: {
+          $route: mockRoute,
+          $router: mockRouter,
+        },
+      },
+      props: {
+        facetList: [
+          { name: "english", count: 1 },
+          { name: "french", count: 2 },
+        ],
+        facetHeader: "language",
+      },
+    });
+
+    wrapper.setData({ checkedFacets: ["english"] });
+
+    expect(wrapper.text()).toMatch("language");
+    expect(wrapper.text()).toMatch("english (1)");
+    expect(wrapper.text()).toMatch("french (2)");
+  });
+
+  it("populates v-model when checkbox is clicked", () => {
```

Comment:
We should test unchecking as well. And possibly some specific testing around the weird bits of arrays / strings that we have to do because of how timdex works.

I would be okay with a new ticket to address these specific cases.

Reply:
Good call. I went ahead and created a ticket on testing: https://mitlibraries.atlassian.net/browse/DISCO-215

There's also a ticket to show checked facets on refresh: https://mitlibraries.atlassian.net/browse/DISCO-210